### ISSUE TITLE
[Bug]: mini-cart hides mobile notification icon

### ISSUE THREAD

**Issue by michaeldcain** [Automattic/wp-calypso#60148]
### Quick summary

On mobile, possibly when the user has a double digit number of drafts, the padding of the mini-cart icon causes the notification icon to not be displayed.

![image (11)](https://user-images.githubusercontent.com/942359/149825633-c15b57ac-2382-43ab-b23b-85700942f8cc.png)

### Steps to reproduce

1. Create a few drafts
2. Add a domain to your cart
3. Visit Calypso on a mobile device

### What you expected to happen

I should be able to see the notifications icon

### What actually happened

I didn't see the notifications icon

### Reproducibility

Consistent

### Severity

Some (< 50%)

### Workaround details

Emptying the user's cart will remove the cart icon (making room for the notifications icon).

**Comment by sirbrillig** [comment]
Good eye. The notifications icon is being pushed off to the right of the page by the padding of each icon, even though the `header` element is using flexbox.

<img width="353" alt="Screen_Shot_2022-01-17_at_2_23_23_PM" src="https://user-images.githubusercontent.com/2036909/149827689-f335ec8d-a57e-4619-ba00-e1fe14a70b51.png">

I think this is mainly because the icons are being spaced with css padding instead of allowing flexbox to do its own spacing. I experimented with my browser and if I set each icon to `padding: 0; flex: 1;` then all the icons become visible.

<img width="323" alt="Screen Shot 2022-01-17 at 3 16 03 PM" src="https://user-images.githubusercontent.com/2036909/149832420-22bbe67e-b8ec-404d-ab6b-a4b8636c6d58.png">

I'll experiment with the actual button source to see if I can find a way to do this in the code.

Another option is that we can hide the shopping cart button at small widths, although that sort of defeats the purpose of having it (giving users a way to get to checkout if they already have items in their cart).

**Comment by shaunandrews** [comment]
Might be worth considering the removal of the cart icon on small screens; It also has the side-effect of displacing the write button, which is intended to remain horizontally centered on small viewports.

**Comment by sirbrillig** [comment]
That's a good point; there's a fundamental issue here which is that if we have six buttons, none of them will be in the center if they are spaced evenly. I've created a PR to hide the cart button for now: https://github.com/Automattic/wp-calypso/pull/60157

I think that perhaps with careful use of flexbox settings or, perhaps more easily with CSS grid, we can align all these buttons correctly at all widths. This will take time to get right, though, so that work can continue beyond this bug fix. I created https://github.com/Automattic/wp-calypso/issues/60158 to track that work.
